pkg/flow/flowtest/commands.go

```diff
@@ -0,0 +1,136 @@
+package flowtest
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/pkg/flow"
+	"github.com/grafana/dskit/backoff"
+	"rsc.io/script"
+)
+
+func createCommands(f *flow.Flow, opts *options) map[string]script.Cmd {
+	cmds := script.DefaultCmds()
+	registerControllerCommands(f, cmds)
+
+	for name, cmd := range opts.cmds {
+		cmds[name] = cmd
+	}
+
+	return cmds
+}
+
+func registerControllerCommands(f *flow.Flow, cmds map[string]script.Cmd) {
+	cmds["controller.load"] = script.Command(
+		script.CmdUsage{
+			Summary: "load a file into the Flow controller",
+			Args:    "file",
+		},
+		func(s *script.State, args ...string) (script.WaitFunc, error) {
+			if len(args) != 1 {
+				return nil, script.ErrUsage
+			}
+
+			bb, err := os.ReadFile(makeAbsolute(s.Getwd(), args[0]))
+			if err != nil {
+				return nil, err
+			}
+			src, err := flow.ParseSource(args[0], bb)
+			if err != nil {
+				return nil, fmt.Errorf("parsing %q: %w", args[0], err)
+			}
+			return nil, f.LoadSource(src, nil)
+		},
+	)
+
+	cmds["controller.start"] = script.Command(
+		script.CmdUsage{
+			Summary: "start the Flow controller",
+			Async:   true,
+			Detail: []string{
+				"This command should almost always be run in the background by appending a &, otherwise the controller will exit immediately after the command finishes'",
```

Comment:
if & is missing, wouldn't the script block execution until controller exits (which could be forever)?